### PR TITLE
Deterministic phf

### PR DESCRIPTION
**Pull request by youknowone**
Let's start from here to ensure not to break things

### CODE REVIEW

**Comment by youknowone** [comment]
@progval do you mind if I use rustfmt to change format?

**Comment by progval** [comment]
Neat thanks.

Could you switch from `ShiftRng` to `rand::rngs::StdRng`? It should work just as well and avoids an extra dep.

What was causing the test build failure before your force-push?

> @progval do you mind if I use rustfmt to change format?

I'll do it after merging this PR

**Comment by youknowone** [comment]
The failure was rand interface issue from test. 
I forgot to port test functions to new version of rand interface 😅

File: generator/Cargo.lock [hunk 1]
```diff
@@ -38,13 +55,49 @@ dependencies = [
...
+]
+
+[[package]]
+name = "rand_xorshift"
+version = "0.3.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "d25bf25ec5ae4a3f1b92f929810509a2f53d7dca2f50b794ff57e3face536c8f"
+dependencies = [
+ "rand_core",
 ]
```
Comment (progval) [inline generator/Cargo.lock:101]: heh, I guess we're pulling it anyway

**Comment by progval** [comment]
This keeps the result constant between two runs on my machine, but I am still getting different from you: https://github.com/progval/unicode_names2/commit/29df74266c7c0e4e4baf80ceded382a8f10baaa3 (though there are some blocks in common, now)

Any idea why?

FWIW I'm using Rustc 1.48.0, Debian 11, on amd64

**Comment by progval** [comment]
Duh, the order is the same only quoting differs. This has to be because of differing Rustc versions

**Comment by progval** [comment]
@youknowone actually I apparently can't run rustfmt on my machine, so go ahead and submit a PR for it

**Pull request merged**